### ISSUE TITLE
Dependencies are getting under the resourceTimeRange

### ISSUE THREAD

**Issue by taauntik** [bryntum/support#8043]
- [x] Try `display:contents` on foreground canvas (investigate all test fails)
- [x] If not possible with `display:contents`, break apart rendering of events vs resourceTimeRanges. Ping Johan for guidance.

[Forum post](https://forum.bryntum.com/viewtopic.php?f=51&t=27314&p=136369#p136369)

Hello Bryntum team,

We are currently upgrading to the latest version of Bryntum and we have noticed that our dependency paths are no longer visible. The  problem is that the dependency paths are behind the resource time ranges which have full opacity. Is there a way around this without adding transparency to the resource time ranges, or is this even a bug ?

In the old version the dependency svg was part of the foreground canvas, like the resource time ranges and had a higher z-index(3). Now it is part of the dependencies-canvas which is behind the foreground-canvas.

Best regards
Thomas

---------------------------------------------------------
Reproducible here https://bryntum.com/products/schedulerpro/examples/constraints/

https://github.com/bryntum/support/assets/75997756/6614e53b-80e9-4b3e-8373-a83978e8b330

With the code below
```javascript
import { SchedulerPro, DateHelper, StringHelper, Tooltip } from '../../build/schedulerpro.module.js?472449';
import shared from '../_shared/shared.module.js?472449';

const resources          = [
        { id : 'r1', name : 'Mike' },
        { id : 'r2', name : 'Linda' },
        { id : 'r3', name : 'Lisa' },
        { id : 'r4', name : 'Madison' },
        { id : 'r5', name : 'Mark' },
        { id : 'r6', name : 'Kate' },
        { id : 'r7', name : 'Dan' },
        { id : 'r8', name : 'Henrik' },
        { id : 'r9', name : 'Rob' },
        { id : 'r10', name : 'Gloria' }
    ],
    events             = [
        {
            id         : 1,
            resourceId : 'r1',
            startDate  : new Date(2019, 0, 1, 8),
            endDate    : new Date(2019, 0, 1, 11),
            name       : 'Investigation',
            iconCls    : 'b-fa b-fa-search'
        }, {
            id         : 2,
            resourceId : 'r1',
            startDate  : new Date(2019, 0, 1, 13),
            endDate    : new Date(2019, 0, 1, 15),
            name       : 'Brief',
            iconCls    : 'b-fa b-fa-newspaper'
        },
        {
            id         : 3,
            resourceId : 'r2',
            startDate  : new Date(2019, 0, 1, 8),
            endDate    : new Date(2019, 0, 1, 9, 30),
            name       : 'Scrum',
            iconCls    : 'b-fa b-fa-bullhorn'
        },
        {
            id         : 4,
            resourceId : 'r3',
            startDate  : new Date(2019, 0, 1, 8),
            endDate    : new Date(2019, 0, 1, 9, 30),
            name       : 'Scrum',
            iconCls    : 'b-fa b-fa-bullhorn'
        },
        {
            id         : 5,
            resourceId : 'r4',
            startDate  : new Date(2019, 0, 1, 7),
            endDate    : new Date(2019, 0, 1, 11),
            name       : 'Meeting',
            iconCls    : 'b-fa b-fa-calendar'
        },
        {
            id         : 6,
            resourceId : 'r4',
            startDate  : new Date(2019, 0, 1, 15),
            endDate    : new Date(2019, 0, 1, 17),
            name       : 'Meeting',
            iconCls    : 'b-fa b-fa-calendar',
            eventColor : 'blue'
        },
        {
            id         : 7,
            resourceId : 'r6',
            startDate  : new Date(2019, 0, 1, 12, 30),
            endDate    : new Date(2019, 0, 1, 19),
            name       : 'Important meeting',
            iconCls    : 'b-fa b-fa-exclamation-triangle',
            eventColor : 'red'
        },
        {
            id         : 8,
            resourceId : 'r6',
            startDate  : new Date(2019, 0, 1, 9),
            endDate    : new Date(2019, 0, 1, 12),
            name       : 'Generic meeting',
            iconCls    : 'b-fa b-fa-calendar'
        },
        {
            id         : 9,
            resourceId : 'r7',
            startDate  : new Date(2019, 0, 1, 9),
            endDate    : new Date(2019, 0, 1, 11),
            name       : 'Dad\'s birthday',
            iconCls    : 'b-fa b-fa-birthday-cake',
            eventColor : 'green'
        },
        {
            id         : 10,
            resourceId : 'r9',
            startDate  : new Date(2019, 0, 1, 13),
            endDate    : new Date(2019, 0, 1, 20),
            name       : 'Golf tournament',
            iconCls    : 'b-fa b-fa-golf-ball',
            eventColor : 'green'
        }
    ],

    resourceTimeRanges = [
        {
            id             : 1,
            resourceId     : 'r1',
            startDate      : '2019-01-01T11:00',
            endDate        : '2019-01-01T13:00',
            name           : 'Lunch',
            // this time range should repeat every day
            recurrenceRule : 'FREQ=DAILY',
            important      : false
        },
        {
            id         : 2,
            resourceId : 'r8',
            startDate  : '2019-01-01T11:00',
            endDate    : '2019-01-01T13:00',
            name       : 'Lunch',
            important  : false
        },
        {
            id         : 3,
            resourceId : 'r9',
            startDate  : '2019-01-01T11:00',
            endDate    : '2019-01-01T13:00',
            name       : 'Lunch',
            important  : false
        },
        {
            id         : 4,
            resourceId : 'r10',
            startDate  : '2019-01-01T11:00',
            endDate    : '2019-01-01T13:00',
            name       : 'Lunch',
            important  : false
        },
        {
            id         : 5,
            resourceId : 'r3',
            startDate  : '2019-01-01T12:00',
            endDate    : '2019-01-01T14:00',
            name       : 'Lunch',
            important  : false
        },
        {
            id           : 6,
            resourceId   : 'r4',
            startDate    : '2019-01-01T12:00',
            duration     : 2,
            durationUnit : 'h',
            name         : 'Lunch',
            important    : false
        },
        {
            id             : 7,
            resourceId     : 'r2',
            startDate      : '2019-01-01T10:00',
            endDate        : '2019-01-01T17:00',
            name           : 'Car maintenance',
            timeRangeColor : 'red',
            important      : true
        },
        {
            id         : 8,
            resourceId : 'r7',
            startDate  : '2019-01-01T12:00',
            endDate    : '2019-01-01T18:00',
            name       : 'Afternoon off (custom style)',
            style      : 'background: rgba(255,165,0,.3);color : orange',
            important  : true
        },
        {
            id         : 9,
            resourceId : 'r5',
            startDate  : '2019-01-01T06:00',
            endDate    : '2019-01-01T20:00',
            name       : 'Parental leave (custom CSS)',
            cls        : 'custom',
            important  : true
        }
    ];
//endregion

let newRangeCount = 0;

const scheduler = new SchedulerPro({
    appendTo : 'container',

    startDate         : new Date(2019, 0, 1, 6),
    endDate           : new Date(2019, 0, 1, 20),
    viewPreset        : 'hourAndDay',
    rowHeight         : 90,
    barMargin         : 3,
    resourceMargin    : 25,
    resourceImagePath : '../_shared/images/users/',

    eventStyle : 'plain',
    eventColor : 'blue',

    columns : [
        { type : 'resourceInfo', text : 'Name', field : 'name', width : 130 }
    ],

    features : {
        resourceTimeRanges : {
            // Enable the resource time range elements to be reachable in the DOM (to show tooltips etc.)
            enableMouseEvents : true
        },
dependencies : true,
    },

    resources,
    events,
    // the store will be used by the "resourceTimeRanges" feature
    resourceTimeRanges,

    // You can use a custom renderer method to output the contents of the range elements. The return value should
    // be a string or a DOMConfig object defining the markup to generate
    resourceTimeRangeRenderer({ resourceTimeRangeRecord, resourceRecord, renderData }) {
        if (resourceTimeRangeRecord.important) {
            // Adds a CSS class to the range element
            renderData.cls.important = 1;

            return [
                {
                    tag   : 'i',
                    class : 'b-fa b-fa-warning'
                },
                {
                    tag  : 'strong',
                    text : `${DateHelper.format(resourceTimeRangeRecord.startDate, 'HH:mm')} - ${DateHelper.format(resourceTimeRangeRecord.endDate, 'HH:mm')} ${resourceTimeRangeRecord.name}`
                }
            ];
        }
        return resourceTimeRangeRecord.name;
    },
    tbar : [
        {
            type     : 'button',
            icon     : 'b-icon-add',
            text     : 'Add new range',
            onAction : () => {
                if (newRangeCount < 10) {
                    scheduler.resourceTimeRangeStore.add({
                        name           : 'New time range',
                        startDate      : new Date(2019, 0, 1, 6),
                        duration       : 2,
                        durationUnit   : 'h',
                        timeRangeColor : 'green',
                        resourceId     : 'r' + (++newRangeCount)
                    });
                }
            }
        },
        {
            type     : 'button',
            icon     : 'b-fa-clock',
            text     : 'Move lunch',
            onAction : () => {
                scheduler.resourceTimeRangeStore.forEach(range => {
                    if (range.name === 'Lunch') {
                        range.startDate = DateHelper.add(range.startDate, 1, 'hour');
                    }
                });
            }
        },
        {
            type       : 'button',
            text       : 'Enable mouse interaction',
            pressed    : true,
            toggleable : true,
            onToggle({ pressed }) {
                scheduler.features.resourceTimeRanges.enableMouseEvents = pressed;
            }
        }
    ],

    listeners : {
        paint() {
            const scheduler = this;

            scheduler.rangeTooltip = new Tooltip({
                rootElement : document.body,
                forSelector : '.b-sch-resourcetimerange',
                getHtml     : ({ event }) => {
                    const
                        rangeRecord = scheduler.resolveResourceTimeRangeRecord(event.target),
                        hours       = DateHelper.as('hour', rangeRecord.duration, rangeRecord.durationUnit);

                    return `${rangeRecord.name} - ${hours} ${DateHelper.getLocalizedNameOfUnit('hour', hours !== 1)}`;
                }
            });
        },

        resourceTimeRangeClick({ resourceTimeRangeRecord }) {
            Toast.show(`You clicked ${resourceTimeRangeRecord.name}`);
        },

        destroy() {
            this.rangeTooltip.destroy();
        }
    }
});
```

**Comment by perryrh0dan** [comment]
Hi, do you have any update regarding the bug fix ? :)

**Comment by matsbryntse** [comment]
@perryrh0dan Working on it, slightly complex to get it fixed and tested but on the way!

**Comment by isglass** [comment]
The fix requires separating resource time ranges and event bars into separate canvases (divs). Changes are on the larger side, the fix will target 6.0.0 (hopefully makes it into alpha-3)
